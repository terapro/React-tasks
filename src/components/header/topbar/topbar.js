import React from 'react';
import PropTypes from 'prop-types';

import {Logo} from 'src/components/common/logo/logo.js';
import {Button} from 'src/components/common/button/button.js';

const TopBar = ({filmMode, setSearchModeCallback}) => (
    <div className={'topbar'}>
      <Logo/>
      {filmMode && <Button content='search' btnWhite btnClick={setSearchModeCallback} />}
    </div>
);

TopBar.propTypes ={
  setSearchModeCallback: PropTypes.func.isRequired,
  filmMode: PropTypes.bool
};
TopBar.defaultProps ={
  filmMode: false
};

export {TopBar};