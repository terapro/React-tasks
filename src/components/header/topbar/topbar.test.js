import React from 'react';
import {TopBar} from 'src/components/header/topbar/topbar.js';
import {shallow} from 'enzyme';

const mockFunc = jest.fn();

describe('<TopBar />', () => {
  describe('Rendering', () => {
    it('should render the component in search mode', () => {
      const wrapper = shallow(<TopBar searchButtonClick={mockFunc} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component in film mode', () => {
      const wrapper = shallow(<TopBar searchButtonClick={mockFunc} filmMode/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});


