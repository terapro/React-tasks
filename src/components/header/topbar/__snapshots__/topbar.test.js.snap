// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> Rendering should render the component in film mode 1`] = `
<div
  className="topbar"
>
  <Logo />
  <Button
    btnBig={false}
    btnBold={false}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={true}
    content="search"
    id=""
  />
</div>
`;

exports[`<TopBar /> Rendering should render the component in search mode 1`] = `
<div
  className="topbar"
>
  <Logo />
</div>
`;

exports[`<TopBar /> Rendering should render the component with default props 1`] = `
<div
  className="topbar"
>
  <Logo />
</div>
`;
